src/lm.jl

```diff
@@ -243,3 +243,46 @@ function confint(obj::LinearModel; level::Real=0.95)
     hcat(coef(obj),coef(obj)) + stderror(obj) *
     quantile(TDist(dof_residual(obj)), (1. - level)/2.) * [1. -1.]
 end
+
+"""
+    cooksdistance(obj::LinearModel)
+Compute Cook's distance for each observation, an estimate of the influence of each data point.
+Currently only model without weights or with normalised weights are supported.
+reference: https://en.wikipedia.org/wiki/Cook%27s_distance
+Credit to Tyler Beason https://github.com/tbeason
+"""
+function cooksdistance(obj::LinearModel)
+    u = residuals(obj)
+    mse = dispersion(obj,true)
+    k = dof(obj)-1
+    d_res = dof_residual(obj)
+    X = modelmatrix(obj)
+    wts = obj.rr.wts
+    if isempty(wts)
+        hii = diag(X * inv(X' * X) * X')
+    else
+        if mean(wts) == 1. # assume that the weights are normalised if the mean is 1 
```

Comment:
What does "normalized" mean in this context? In GLM.jl weights are frequency weights (i.e. integers most of the time), so their mean should probably never be 1. This differs from e.g. R.

Reply:
I see, well I assumed that the weights were weights and not frequencies.

Also the degree of freedom function seems to use weights as weights and not frequencies:
`dof(x::LinearModel) = length(coef(x)) + 1` (line 159 in lm.jl) in case of frequencies then this should be the sum of the frequencies there, right? I realised that I have be confused between weights and frequencies.

If the caller is thinking about weights, then I assumed it would be relatively simple for the caller to normalized the weights and since the beginning of the code was already done by the original author. 

Now if we want to be strict that they are frequencies, then I am afraid my analytical skills are not up to the work of finding the adequate calculation. I would just put `error` on line 264.

Let me know your preferred way.

Reply:
> Also the degree of freedom function seems to use weights as weights and not frequencies:
> `dof(x::LinearModel) = length(coef(x)) + 1` (line 159 in lm.jl) in case of frequencies then this should be the sum of the frequencies there, right? I realised that I have be confused between weights and frequencies.

`dof` gives the number of *consumed* degrees of freedom. The number of *residual* DOF is given by `dof_residual`, which is computed as `nobs(m) - dof(m) + 1`. `nobs` is in turn defined as the sum of weights:
https://github.com/JuliaStats/GLM.jl/blob/b39253f18e5d370c15ca8c7ac8d0f09c2dc04373/src/linpred.jl#L250

> Now if we want to be strict that they are frequencies, then I am afraid my analytical skills are not up to the work of finding the adequate calculation. I would just put `error` on line 264.

Usually frequency weights are the easier kind of weights to reason about since the results should be exactly equivalent to fitting the same model without weights but on data with each row repeated a number of times equal to its weight. But we would have to decide whether the distance for a given row should represent the effect of removing one observation corresponding to this row, or all of them (probably the former makes more sense). Anyway if you don't feel like implementing this we can just throw an error in the presence of weights for now.
if you don't 

Cc: @tbeason

Reply:
@nalimilan, I understand the frequencies the same way you do. Although I am not sure there is any for us to decide with regards to the Cooks's distance as the results should match what is present in other packages (like SAS). I want to implement it but I realized that I am laking the skills to do it correctly. If there is a known formula let me know and I will plug it in.
If it is helpful I could add the tests for the frequencies, it is not an issue.

Reply:
Let's just throw an error in the presence of weights if we don't know the formula.

Reply:
done